### PR TITLE
adds some new coalition weapons

### PR DESCRIPTION
**Pull request by crunchypretzels**
image

Adds four new weapons from the United Syndicates and the Federal Technocracy.

Himeo has received two new guns based on the plasma cutter; the Guthrie assault blaster and the Sabo-Tabby heavy blaster pistol. Both use a hydrogen cell 'magazine' to mimic the more advanced, more accurate Zavodskoi blasters on the market.

Galatea, likewise, has received the O61-B laser rifle pack set, an export model of their own O61 Infantry Laser Rifle, and a gauss 'nitro express' rifle, which fires tungsten slugs. Attempting to fire the O61 without a Galatean implant will result in consequences.

All sprites are my own handiwork, and all credit for the laser pack goes to Alberyk for his work on the gatling gun. Matt and Schwann also provided a great deal of help.

### CODE REVIEW

**Comment by crunchypretzels** [comment]
!wip

**Comment by crunchypretzels** [comment]
!review

File: code/game/objects/items/weapons/implants/implants/galatean.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<b>Life:</b> Six years.<BR>
+<b>Important Notes:</b> Personnel injected with this device can operate certain approved models of laser weapon without side effects..<BR>
+<HR>
+<b>Implant Details:</b><BR>
+<b>Function:</b> Conducts the body's electricity into usable forms of electricity for Galatean export-model weaponry.<BR>
+<b>Integrity:</b> Implant will last for a period of six standard years, after which it will degrade and require removal."}
+
+/obj/item/implant/export_lasgun/implanted(mob/M)
+	to_chat(M, SPAN_GOOD("Static dances across your skin."))
+	if(istype(M, /mob/living/carbon/human))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/implants/implants/galatean.dm:27]: use ishuman

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -509,3 +509,167 @@
...
+		gun.forceMove(src)
+		update_icon()
+
+	armed = FALSE
+	return
+
+/obj/item/galatealaserpack/Destroy()
+	if(gun)
+		qdel(gun)
+		gun = null
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:607]: QDEL_NULL(gun)

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -509,3 +509,167 @@
 	fire_delay = 20
 	accuracy = 40
 	muzzle_flash = 10
+
+/obj/item/galatealaserpack
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:513]: use a clearer path name, 3 words with no spacing is quite a lot

File: code/modules/projectiles/guns/energy/special.dm [hunk 4]
```diff
@@ -509,3 +509,167 @@
...
+/obj/item/galatealaserpack/ui_action_click()
+	if(src in usr)
+		toggle_lasergun()
+
+/obj/item/galatealaserpack/verb/toggle_lasergun() // ui button and a verb
+	set name = "Deploy the Laser Rifle"
+	set category = "Object"
+	var/mob/living/carbon/human/user
+	if(istype(usr,/mob/living/carbon/human))
+		user = usr
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:555]: ishuman

File: code/modules/projectiles/guns/energy/special.dm [hunk 5]
```diff
@@ -509,3 +509,167 @@
...
+	else
+		return
+
+	if(!user)
+		return
+
+	if (user.back!= src)
+		to_chat(user, "<span class='warning'>\The [src] must be worn to deploy \the [gun]!</span>")
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:563]: use the SPAN_X macros, needs to be done throughout this file

File: code/modules/projectiles/guns/energy/special.dm [hunk 6]
```diff
@@ -509,3 +509,167 @@
...
+	return
+
+/obj/item/galatealaserpack/Destroy()
+	if(gun)
+		qdel(gun)
+		gun = null
+	return ..()
+
+/obj/item/galatealaserpack/attackby(obj/item/attacking_item, mob/user, params) // if you hit the pack with the gun, "stow" it
+	if(attacking_item == gun)
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:611]: use_check_and_message this as well

File: code/modules/projectiles/guns/energy/special.dm [hunk 7]
```diff
@@ -509,3 +509,167 @@
...
+		remove_gun()
+		update_icon()
+		user.update_inv_back()
+		return 1
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/energy/special.dm:615]:
```suggestion
		return TRUE
```

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**